operator/podinfo/podinfo_controller_test.go

```diff
@@ -0,0 +1,376 @@
+		})
+
+		t.Run("Pod labels not available", func(t *testing.T) {
+			pod := randomPodGenerator()
+			pod.Labels = nil
+			assert.False(t, checkIfReady(pod), "Pod Labels not available still returning Pod to be ready")
+		})
+
+		t.Run("Pod annotations not available", func(t *testing.T) {
+			pod := randomPodGenerator()
+			pod.Annotations = nil
+			assert.False(t, checkIfReady(pod), "Pod Annotations not available still returning Pod to be ready")
+		})
+
+		t.Run("Pod UID not available", func(t *testing.T) {
+			pod := randomPodGenerator()
+			pod.UID = ""
+			assert.False(t, checkIfReady(pod), "Pod UID not available still returning Pod to be ready")
+		})
+	})
+}
+
+func TestCheckIfChanged(t *testing.T) {
+	t.Run("Check if the pod and podInfo are different", func(t *testing.T) {
+
+		// All fields match
+		t.Run("All fields same", func(*testing.T) {
+			pod := randomPodGenerator()
+			podInfo := getPodInfo(pod)
+			assert.False(t, checkIfChanged(pod, podInfo), "All fields match, still returning pod to be changed")
+		})
+
+		// Fields mismatch
+		t.Run("Name not available", func(t *testing.T) {
+			pod := randomPodGenerator()
+			podInfo := getPodInfo(pod)
+			pod.Name = getRandString(getRandNum())
+
+			assert.True(t, checkIfChanged(pod, podInfo), "Pod Name changed, still returning pod not changed")
+		})
+
+		t.Run("Namespace of the pod changed", func(t *testing.T) {
+			pod := randomPodGenerator()
+			podInfo := getPodInfo(pod)
+			pod.Namespace = getRandString(getRandNum())
+
+			assert.True(t, checkIfChanged(pod, podInfo), "Pod Namespace changed, still returning pod not changed")
+		})
+
+		t.Run("Pod IP of the pod changed", func(t *testing.T) {
+			pod := randomPodGenerator()
+			podInfo := getPodInfo(pod)
+			pod.Status.PodIP = getRandIP()
+			assert.True(t, checkIfChanged(pod, podInfo), "Pod IP changed, still returning pod not changed")
+		})
+
+		//t.Run("Pod IPs changed", func(t *testing.T) {
+		//	pod := randomPodGenerator()
+		//	podInfo := getPodInfo(pod)
+		//	randIP, randIPs := getRandIPs()
+		//
+		//	randPodIPs := make([]corev1.PodIP, len(randIPs))
+		//	for i, IP := range randIPs {
+		//		randPodIPs[i] = corev1.PodIP{IP: IP}
+		//	}
+		//	pod.Status.PodIP = randIP
+		//	pod.Status.PodIPs = randPodIPs
+		//
+		//	assert.True(t, checkIfChanged(pod, podInfo), "Pod IPs changed, still returning pod not changed")
+		//})
+
+		//t.Run("Pod Labels changed", func(t *testing.T) {
+		//	pod := randomPodGenerator()
+		//	podInfo := getPodInfo(pod)
+		//	pod.Labels = getRandMap()
+		//	assert.True(t, checkIfChanged(pod, podInfo), "Pod Labels changed, still returning pod not changed")
+		//})
+		//
+		//t.Run("Pod annotations changed", func(t *testing.T) {
+		//	pod := randomPodGenerator()
+		//	podInfo := getPodInfo(pod)
+		//	pod.Annotations = getRandMap()
+		//	assert.True(t, checkIfChanged(pod, podInfo), "Pod Annotations changed, still returning pod not changed")
+		//})
+
```

Comment:
```suggestion
```

Reply:
Commented it out, because I found a corner case, need to fix that and uncomment.